src/utils.ts

```diff
@@ -1,34 +1,75 @@
-export function splitPath (path: string): number[] {
-	const pathArray = [];
+/**
+ * Parse a Bip32 HD Path into an array of elements.
+ *
+ * @param {string} path Bip32 Path string
+ * @returns {number[]} array of parsed Bip32 elements
+ * @throws {Error} if the path string-length is 0
+ */
+export function parseBip32Path(path: string): number[] {
+	if (path.length === 0) {
+		throw new Error("Invalid Path Length");
+	}
+
+	const elements = [];
 
-	for (const entry of path.split("/")) {
-		let number = parseInt(entry, 10);
+	for (const element of path.split("/")) {
+		let number = parseInt(element, 10);
 
 		if (isNaN(number)) {
-			throw new Error("Invalid path provided");
+			throw new Error("Invalid Path");
 		}
 
-		if (entry.length > 1 && entry[entry.length - 1] === "'") {
+		if (element.length > 1 && element[element.length - 1] === "'") {
 			number += 0x80000000;
 		}
 
-		pathArray.push(number);
+		elements.push(number);
 	}
 
-	return pathArray;
+	return elements;
 }
 
-export function writePathsToBuffer (paths: number[], buffer: Buffer): void {
-	buffer[0] = paths.length;
-
-	for (const index in paths) {
-		buffer.writeUInt32BE(paths[index], 1 + (4 * parseInt(index, 10)));
+/**
+ * Write a parsed Bip32 element array a Buffer.
+ *
+ * @param {number[]} elements array of parsed Bip32 elements
+ * @param {Buffer} buffer destination where Bip32 elements will be written
+ * @throws {Error} if the path length is 0 or greater than 10
+ */
+export function writeBip32ElementsToBuffer(
+	elements: number[],
+	buffer: Buffer
+): void {
+	if (elements.length === 0 || elements.length > 10) {
+		throw new Error("Invalid Path Length");
 	}
+
+	buffer[0] = elements.length;
+
+	elements.forEach(function (element, index) {
```

Comment:
We tend to stick to for/in or for/of loops as those are generally faster than forEach